### PR TITLE
New, very simple data sources for GCR Repo and Image.

### LINKED ISSUE #607: GCP container-registry
Hi Team, 

It would be great if we can have provider for container-registry

https://cloud.google.com/container-registry/

Thanks.

### PR DESCRIPTION
**Pull request by ndmckinley**
I think this fixed #607, but I'm a little fuzzy on the use-case there, so I'd like someone who requested / thumbs-up'd that to look at the code and make sure it's helpful before I invest the time in writing the docs.

### CODE REVIEW

**Comment by mahmoudian1** [comment]
Thanks @ndmckinley looks good

File: google/provider.go [hunk 1]
```diff
@@ -75,7 +76,8 @@ func Provider() terraform.ResourceProvider {
 			"google_compute_region_instance_group": dataSourceGoogleComputeRegionInstanceGroup(),
 			"google_container_cluster":             dataSourceGoogleContainerCluster(),
 			"google_container_engine_versions":     dataSourceGoogleContainerEngineVersions(),
-			"google_active_folder":                 dataSourceGoogleActiveFolder(),
+			"google_gcr_repository":                dataSourceGoogleGcrRepo(),
```
Comment (rosbo) [inline google/provider.go]: should we use the acronym GCR or expand it to `container_registry`.

It is also weird to have google_g(oogle)c(ontainer)r(egistry)_repository. `google` is repeated twice.

Comment (ndmckinley) [reply google/provider.go]: Agreed - expanded it.

File: google/data_source_google_gcr_repository_test.go [hunk 2]
```diff
@@ -9,7 +9,7 @@ import (
 func TestDataSourceGoogleGcrRepository(t *testing.T) {
```
Comment (rosbo) [inline google/data_source_google_gcr_repository_test.go]: Rename the tests too

File: google/data_source_google_gcr_repository.go [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+			},
+			"image_url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func gcrRepoRead(d *schema.ResourceData, meta interface{}) error {
```
Comment (rosbo) [inline google/data_source_google_gcr_repository.go]: Rename these

**Comment by ndmckinley** [comment]
For reasons I can't figure out, these new docs are crashing the website.  I know it's got to be these because it doesn't happen on master, and it does happen on this branch, even with the branch rebased on top of master.  The error message is not helping me:

```
NoMethodError: undefined method `force_encoding' for nil:NilClass              
        /usr/local/bundle/gems/tilt-1.4.1/lib/tilt/template.rb:289:in `ensure in binary'                                                                      
        /usr/local/bundle/gems/tilt-1.4.1/lib/tilt/template.rb:289:in `binary' 
        /usr/local/bundle/gems/tilt-1.4.1/lib/tilt/template.rb:279:in `extract_magic_comment'                                                                 
        /usr/local/bundle/gems/tilt-1.4.1/lib/tilt/template.rb:275:in `extract_encoding'                                                                      
        /usr/local/bundle/gems/tilt-1.4.1/lib/tilt/template.rb:191:in `precompiled'                                                                           
        /usr/local/bundle/gems/tilt-1.4.1/lib/tilt/erb.rb:57:in `precompiled'  
        /usr/local/bundle/gems/tilt-1.4.1/lib/tilt/erb.rb:104:in `precompiled' 
        /usr/local/bundle/gems/tilt-1.4.1/lib/tilt/template.rb:245:in `compile_template_method'                                                               
        /usr/local/bundle/gems/tilt-1.4.1/lib/tilt/template.rb:240:in `compiled_method'                                                                       
        /usr/local/bundle/gems/tilt-1.4.1/lib/tilt/template.rb:169:in `evaluate'                                                                              
        /usr/local/bundle/gems/tilt-1.4.1/lib/tilt/template.rb:103:in `render' 
        /usr/local/bundle/gems/middleman-core-3.4.1/lib/middleman-core/core_extensions/rendering.rb:313:in `render_individual_file'                           
        /usr/local/bundle/gems/middleman-core-3.4.1/lib/middleman-core/core_extensions/rendering.rb:153:in `render_template'                                  
        /usr/local/bundle/gems/middleman-core-3.4.1/lib/middleman-core/sitemap/resource.rb:127:in `block in render'                                           
        /usr/local/bundle/gems/activesupport-4.2.8/lib/active_support/notifications.rb:166:in `instrument'                                                    
        /usr/local/bundle/gems/middleman-core-3.4.1/lib/middleman-core/util.rb:41:in `instrument'                                                             
        /usr/local/bundle/gems/middleman-core-3.4.1/lib/middleman-core/application.rb:244:in `instrument'                                                     
        /usr/local/bundle/gems/middleman-core-3.4.1/lib/middleman-core/sitemap/resource.rb:14:in `instrument'                                                 
        /usr/local/bundle/gems/middleman-core-3.4.1/lib/middleman-core/sitemap/resource.rb:100:in `render'                                                    
        /usr/local/bundle/gems/middleman-core-3.4.1/lib/middleman-core/core_extensions/request.rb:260:in `process_request'                                    
        /usr/local/bundle/gems/middleman-core-3.4.1/lib/middleman-core/core_extensions/request.rb:210:in `block in call!'                                     
        /usr/local/bundle/gems/middleman-core-3.4.1/lib/middleman-core/core_extensions/request.rb:209:in `catch'                                              
        /usr/local/bundle/gems/middleman-core-3.4.1/lib/middleman-core/core_extensions/request.rb:209:in `call!'                                              
        /usr/local/bundle/gems/middleman-core-3.4.1/lib/middleman-core/core_extensions/request.rb:195:in `call'                                               
        /usr/local/bundle/gems/rack-1.6.8/lib/rack/urlmap.rb:66:in `block in call'                                                                            
        /usr/local/bundle/gems/rack-1.6.8/lib/rack/urlmap.rb:50:in `each'      
        /usr/local/bundle/gems/rack-1.6.8/lib/rack/urlmap.rb:50:in `call'      
        /usr/local/bundle/gems/rack-livereload-0.3.16/lib/rack/livereload.rb:23:in `_call'                                                                    
        /usr/local/bundle/gems/rack-livereload-0.3.16/lib/rack/livereload.rb:14:in `call'                                                                     
        /usr/local/bundle/gems/rack-1.6.8/lib/rack/showexceptions.rb:24:in `call'                                                                             
        /usr/local/bundle/gems/rack-1.6.8/lib/rack/head.rb:13:in `call'        
        /usr/local/bundle/gems/rack-1.6.8/lib/rack/lint.rb:49:in `_call'       
        /usr/local/bundle/gems/rack-1.6.8/lib/rack/lint.rb:37:in `call'        
        /usr/local/bundle/gems/rack-1.6.8/lib/rack/builder.rb:153:in `call'    
        /usr/local/bundle/gems/rack-1.6.8/lib/rack/handler/webrick.rb:88:in `service'                                                                         
        /usr/local/lib/ruby/2.3.0/webrick/httpserver.rb:140:in `service'       
        /usr/local/lib/ruby/2.3.0/webrick/httpserver.rb:96:in `run'            
        /usr/local/lib/ruby/2.3.0/webrick/server.rb:296:in `block in start_thread'
```

Any ideas?

**Comment by rosbo** [comment]
Rename also the go files.

File: website/docs/d/google_container_registry_repository.html.markdown [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+sidebar_current: "docs-google-datasource-container-repo"
+description: |-
+  Get URLs for a given project's container registry repository.
+---
+
+# google\_container\_registry\_repository
+
+This data source fetches the project name, and provides the appropriate URLs to use for container registry for this project.
+
+The URLs are computed entirely offline - as long as the project exists, they will be valid, but this data source does not contact GCR at any point.
```
Comment (rosbo) [inline website/docs/d/google_container_registry_repository.html.markdown]: If you want to use the GCR acronym, you should spell it out fully the first time you use it and add the acronym between parentheses after it.

File: google/data_source_container_registry.go [hunk 5]
```diff
@@ -0,0 +1,105 @@
+package google
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceGoogleContainerRepo() *schema.Resource {
```
Comment (rosbo) [inline google/data_source_container_registry.go]: I would split the 2 data sources in different files to follow our convention of one resource or one data source per source file

**Comment by ndmckinley** [comment]
I take back what I said about the website - error repros when I delete both the files I added and use the google.erb from master, so this change probably doesn't cause that issue.

File: google/data_source_container_registry.go (hunk 5 above)
Comment (ndmckinley) [reply google/data_source_container_registry.go]: Done.

**Comment by ndmckinley** [comment]
```
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./google -v -run=TestDataSourceGoogleContainer -timeout 120m
=== RUN   TestDataSourceGoogleContainerRegistryRepository
=== RUN   TestDataSourceGoogleContainerRegistryImage
--- PASS: TestDataSourceGoogleContainerRegistryRepository (0.06s)
--- PASS: TestDataSourceGoogleContainerRegistryImage (0.08s)
PASS
ok      github.com/terraform-providers/terraform-provider-google/google 0.086s
```

```
02:18 $ make vet
go vet .
✔
```

**Pull request merged**